NumLib/ODESolver/NonlinearSolver.h

```diff
@@ -108,11 +110,18 @@ class NonlinearSolver<NonlinearSolverTag::Newton> final
 
     void assemble(GlobalVector const& x) const override;
 
+    void calculateOutOfBalanceForces(GlobalVector const& x) override;
```

Comment:
It is better to move it to Process. In future, we can use CRTP idiom to encapsulate all mechanical process related stuff.